src/realm_dart_sync.cpp

```diff
@@ -0,0 +1,140 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#include <realm/object-store/sync/sync_session.hpp>
+#include <realm/sync/config.hpp>
+#include "realm_dart.hpp"
+#include "realm_dart_sync.h"
+
+RLM_API void realm_dart_http_request_callback(realm_userdata_t userdata, const realm_http_request_t request, void* request_context) {
+    // the pointers in error are to stack values, we need to make copies and move them into the scheduler invocation
+    struct request_copy_buf {
+        std::string url;
+        std::string body;
+        std::map<std::string, std::string> headers;
+        std::vector<realm_http_header_t> headers_vector;
+    } buf;
+
+    realm_http_request_t request_copy = request; // copy struct
+
+    buf.url = request.url;
+    request_copy.url = buf.url.c_str();
+    buf.body = std::string(request.body, request.body_size);
+    request_copy.body = buf.body.data();
+
+    buf.headers_vector.reserve(request.num_headers);
```

Comment:
Core is not using this. I suggest we update core as well if we really thing it makes a difference

Reply:
Core uses the headers map in the request to make authenticated REST calls using the `Authorization` header.

Reply:
but unrelated to this PR

Reply:
this is what @nielsenko suggested before, so lets leave it here. Would like Core to use this as well